src/helpers/fetches.js

```diff
@@ -0,0 +1,74 @@
+import log from 'cozy-logger'
+import { isEqual } from 'lodash'
+import { DOCTYPE_CONTACTS } from './doctypes'
+import { updateIndexFullNameAndDisplayName } from './contacts'
+
+/**
+ * Fetch and returns a promise of contacts to update according to this :
+ * The update date of the contact is more recent than the 'date'
+ * @param {object} client - cozyClient
+ * @param {Date} date - date of comparison
+ * @returns {promise} contacts to update
+ */
+export const fetchContactsToUpdate = async (client, date) => {
+  try {
+    const dateUTCForced = new Date(date).toISOString()
+    const contacts = []
+
+    const queryDef = client
+      .find(DOCTYPE_CONTACTS)
+      .where({
+        trashed: {
+          $exists: false
+        },
+        $or: [
+          {
+            cozyMetadata: { $exists: false }
+          },
+          {
+            'cozyMetadata.updatedAt': { $gt: dateUTCForced }
+          }
+        ]
+      })
+      .indexFields(['_id'])
+      .limitBy(1000)
+
+    const result = await client.queryAll(queryDef)
+    const expected = result.map(contact =>
+      updateIndexFullNameAndDisplayName(contact)
+    )
+    result.map((contact, index) => {
+      if (!isEqual(contact, expected[index])) return contacts.push(contact)
+    })
+
+    log('info', `found ${contacts.length} contact(s) to update`)
+
+    return contacts
+  } catch (e) {
+    throw new Error(`Can't find elements : ${e}`)
+  }
+}
+
+/**
+ * Fetch and returns a promise of service's last success time
+ * @param {object} client - cozyClient
+ * @param {string} serviceName - name of the service
+ * @returns {promise} last success time of a service
```

Comment:
```suggestion
 * @returns {promise<Date>} last success time of a service
```

Date or String ? I don't know

Reply:
String :)